coreneuron/sim/fadvance_core.cpp

```diff
@@ -108,11 +108,36 @@ void finalize_progress_bar() {
     }
 }
 
-void nrn_fixed_step_group_minimal(int n) {
-    static int step = 0;
+void nrn_fixed_single_steps_minimal(int total_sim_steps, double tstop) {
+    const int progressbar_update_interval = 5;
+    static int current_steps = 0;
+    initialize_progress_bar(total_sim_steps);
+#if NRNMPI
+    double updated_tstop = tstop - dt;
+    nrn_assert(nrn_threads->_t <= tstop);
+    // It may very well be the case that we do not advance at all
+    while (nrn_threads->_t <= updated_tstop) {
+#else
+    double updated_tstop = tstop - .5 * dt;
+    while (nrn_threads->_t < updated_tstop) {
+#endif
+        nrn_fixed_step_minimal();
+        if (stoprun) {
+            break;
+        }
+        current_steps++;
+        if (current_steps%progressbar_update_interval) {
```

Comment:
```suggestion
        if (current_steps % progressbar_update_interval == 0) {
```
otherwise this would be true for all `current_steps` except the ones divisible by 5? I suppose you want to update the progressbar only every 5 steps?
Also, if you really prefer:
```suggestion
        if (!(current_steps % progressbar_update_interval)) {
```

Reply:
Good catch! This is what I wanted to do. I just forgot the ` == 0`.. The second suggestion is even better. Thanks a lot :+1: